pkg/cluster/k8sres.go

```diff
@@ -1896,7 +1896,19 @@ func (c *Cluster) generateCloneEnvironment(description *acidv1.CloneDescription)
 
 		if description.S3SecretAccessKey != "" {
 			result = append(result, v1.EnvVar{Name: "CLONE_AWS_SECRET_ACCESS_KEY", Value: description.S3SecretAccessKey})
-		}
+		} else if description.S3AccessSecretName != "" {
```

Comment:
This code here deals with cloning. `S3AccessSecretName` is not a field of type [CloneDescription](https://github.com/zalando/postgres-operator/blob/7c878f8aaab9bee1f86f2022fa89705c53da77e0/pkg/apis/acid.zalan.do/v1/postgresql_type.go#L192). Lets focus only on logical backup here.